app/uk/gov/hmrc/exports/controllers/AmendmentController.scala

```diff
@@ -36,18 +36,29 @@ class AmendmentController @Inject() (
 )(implicit executionContext: ExecutionContext)
     extends RESTController(cc) with JSONResponses {
 
-  val create: Action[SubmissionAmendment] = authenticator.authorisedAction(parsingJson[SubmissionAmendment]) { implicit request =>
+  val handler: Action[SubmissionAmendment] = authenticator.authorisedAction(parsingJson[SubmissionAmendment]) { implicit request =>
+    val submissionAmendment = request.body
+    val declarationId = submissionAmendment.declarationId
     metrics
-      .timeAsyncCall(Timers.amendmentUpdateDeclarationStatusTimer)(submissionService.markCompleted(request.eori, request.request.body.declarationId))
+      .timeAsyncCall(Timers.amendmentUpdateDeclarationStatusTimer)(submissionService.markCompleted(request.eori, declarationId))
       .flatMap {
-        case Some(declaration) if declaration.isCompleted => Future.successful(Conflict("Amendment has already been submitted."))
+        case Some(declaration) if declaration.isCompleted =>
+          val message = s"Amendment for declaration(${declarationId}) has already been submitted."
+          Future.successful(Conflict(message))
 
         case Some(declaration) if declaration.declarationMeta.status != AMENDMENT_DRAFT =>
-          Future.successful(Conflict(s"Attempted to submit declaration with status ${declaration.declarationMeta.status} as an amendment."))
+          val message = s"Attempted to submit declaration(${declarationId}) with status ${declaration.declarationMeta.status} as an amendment."
+          Future.successful(Conflict(message))
 
-        case Some(declaration) => submissionService.amend(request.eori, request.body, declaration).map(Ok(_))
+        case Some(declaration) =>
+          if (submissionAmendment.isCancellation) {
+            if (submissionAmendment.fieldPointers.size != 1 || !submissionAmendment.fieldPointers.head.isBlank) {
```

Comment:
Why are we checking both `isCancellation` and fieldPointer is 1 or blank (should not isCacellation be enough)? :thinking:

Reply:
Yep, it should be enough. I was just too conservative probably. I can remove it btw. Not a big deal